microsoft/PowerToys
# [PT Run] Improve the Win32 Program Indexing speed

## Summary of the Pull Request

**What is this about:**
Improve the performance of the Win32 scanning.

**What is include in the PR:** 
- Performance Improvements
  - Parallelize collection of all paths with multiple sources
  - Deduplicate all paths which are duplicate   
  - Resolve all indexed program paths in parallel 
    -  When checking Steam/Epic games, do not read whole file if not needed
- Refactored indexing on several points
  - Always LowerCase the fullpath (like done on other places
  - Updated some usage of more modern C# features where i think the code became more readable
- Removed Indexing from constructor (so it won't run if the plugin is disabled)
- Fix Chrome PWA's not having the correct AppType
- Change Security Exceptions to Warnings in the logs 
-  ~~Allow Searching by executable name (e.g. WINWORD.EXE) exluded as this changes behavior~~

**How does someone test / validate:** 

- Run PT Run and check the logs if it's faster for that device
- Disable PT Run within the Settings UI & the 'Microsoft.Plugin.Program.Main - Win32Program index cost ' will not be logged

## Quality Checklist

- [x] **Linked issue:** #7208

## Contributor License Agreement (CLA)
A CLA must be signed. If not, go over [here](https://cla.opensource.microsoft.com/microsoft/PowerToys) and sign the CLA.


# Additional Information
